Subject: [PATCH] Current view findFocus maybe null, which cause NullPointerException. 
When run monkey test on TabHost application, easily reproduce the NullPointerException. 

diff --git a/core/java/android/widget/TabHost.java b/core/java/android/widget/TabHost.java
index 238dc55..22b66a2 100644
--- a/core/java/android/widget/TabHost.java
+++ b/core/java/android/widget/TabHost.java
@@ -340,8 +340,11 @@ public boolean dispatchKeyEvent(KeyEvent event) {
                     soundEffect = SoundEffectConstants.NAVIGATION_UP;
                     break;
             }
+
+            View focusView = mCurrentView.findFocus();
+
             if (event.getKeyCode() == keyCodeShouldChangeFocus
-                    && mCurrentView.findFocus().focusSearch(directionShouldChangeFocus) == null) {
+                    && (focusView == null || focusView.focusSearch(directionShouldChangeFocus) == null)) {
                 mTabWidget.getChildTabViewAt(mCurrentTab).requestFocus();
                 playSoundEffect(soundEffect);
                 return true;
